### PR TITLE
feat: restructure cli

### PR DESCRIPTION
**Pull request by crqra**
This PR extracts the root command to it's own package, adds an `exec` command that has the same functionality as the root command had. The root command defaults to `exec` if no other command is specified.

Rename the program to `rsr` for simplicity but keeps the `reposaur` name as well in new releases. The way used to achieve this was adding a second build in the goreleaser config but this actually generates a completely different build (in the sense it's diff directories) which I don't particularly like. Shouldn't we just remove the `reposaur` altogether to avoid confusion in the future and have this settled now in the early stage?

The default client being created in the SDK constructor has been removed to the root cmd package. The SDK will still create a default client using the `util.GitHubTransport` (made public) if none is specified.

### CODE REVIEW

**Comment by crqra** [comment]
@jsmvaldivia WDYT about removing the `reposaur` bin altogether?

File: pkg/cmd/exec/exec.go [hunk 1]
```diff
@@ -8,44 +8,42 @@ import (
...
 
 var cmd = &cobra.Command{
-	Use:   "reposaur",
+	Use:   "exec",
 	Short: "Executes a set of Rego policies against the data provided",
 	Long:  "Executes a set of Rego policies against the data provided",
 }
 
-func NewCommand() *cobra.Command {
+func NewCommand(f *cmdutil.Factory) *cobra.Command {
```
Comment (jsmvaldivia) [inline pkg/cmd/exec/exec.go:30]: is this factory being used? cannot find where..

Comment (crqra) [reply pkg/cmd/exec/exec.go:30]: Yes, here: https://github.com/reposaur/reposaur/pull/13/files/7c7bdb74f820afdeb70c03e2645c78750fb52d7c#diff-0b23fbb550d65237bf314a33bea972602305a7bd7572b94a191fa732cf8ae5b7R44

Comment (crqra) [reply pkg/cmd/exec/exec.go:30]: L44 on the new changes

Comment (jsmvaldivia) [reply pkg/cmd/exec/exec.go:30]: sry was #blind

**Review by jsmvaldivia — APPROVED** [review]
LGTM! I was actually wondering if we should create a new go module here

**Pull request merged**